src/tools/Reporting/Reporting/Reporter.cs

```diff
@@ -0,0 +1,119 @@
+        private List<Test> tests = new List<Test>();
+        protected IEnvironment environment;
+
+        private Reporter() { }
+
+        public void AddTest(Test test)
+        {
+            if (tests.Any(t => t.Name.Equals(test.Name)))
+                throw new Exception($"Duplicate test name, {test.Name}");
+            tests.Add(test);
+        }
+
+        /// <summary>
+        /// Get a Reporter. Relies on environment variables.
+        /// </summary>
+        /// <param name="environment">Optional environment variable provider</param>
+        /// <returns>A Reporter instance or null if the environment is incorrect.</returns>
+        public static Reporter CreateReporter(IEnvironment environment = null)
+        {
+            var ret = new Reporter();
+            ret.environment = environment == null ? new EnvironmentProvider() : environment;
+            if (!ret.CheckEnvironment())
+            {
+                return null;
+            }
+                        
+            ret.Init();
+            return ret;
+        }
+
+        private void Init()
+        {
+            run = new Run
+            {
+                CorrelationId = environment.GetEnvironmentVariable("HELIX_CORRELATION_ID"),
+                PerfRepoHash = environment.GetEnvironmentVariable("PERFLAB_PERFHASH"),
+                Name = environment.GetEnvironmentVariable("PERFLAB_RUNNAME"),
+                Queue = environment.GetEnvironmentVariable("PERFLAB_QUEUE"),
+            };
+            Boolean.TryParse(environment.GetEnvironmentVariable("PERFLAB_HIDDEN"), out bool hidden);
+            run.Hidden = hidden;
+            var configs = environment.GetEnvironmentVariable("PERFLAB_CONFIGS");
+            if (!String.IsNullOrEmpty(configs)) // configs should be optional.
+            {
+                foreach (var kvp in configs.Split(';'))
+                {
+                    var split = kvp.Split('=');
+                    run.Configurations.Add(split[0], split[1]);
+                } 
+            }
+
+            os = new Os()
+            {
+                Name = $"{RuntimeEnvironment.OperatingSystem} {RuntimeEnvironment.OperatingSystemVersion}",
+                Architecture = RuntimeInformation.OSArchitecture.ToString(),
+                Locale = CultureInfo.CurrentUICulture.ToString()
+            };
+
+            build = new Build
+            {
+                Repo = environment.GetEnvironmentVariable("PERFLAB_REPO"),
+                Branch = environment.GetEnvironmentVariable("PEFRFLAB_BRANCH"),
+                Architecture = environment.GetEnvironmentVariable("PERFLAB_BUILDARCH"),
+                Locale = environment.GetEnvironmentVariable("PERFLAB_LOCALE"),
+                GitHash = environment.GetEnvironmentVariable("PERFLAB_HASH"),
+                BuildName = environment.GetEnvironmentVariable("PERFLAB_BUILDNUM"),
+                TimeStamp = DateTime.Parse(environment.GetEnvironmentVariable("PERFLAB_BUILDTIMESTAMP")),
```

Comment:
I feel like this would be easier if we figured out the timestamp in-code, if we are going to use git to tell us the timestamp.

Reply:
We will - I'm going to put it in benchmarks_ci.py for now, and in the future when we have the build information on the helix build machine we will figure it out there. I don't want the reporting library calling out to git itself.

Reply:
Take a look at `build.py` from the BenchView scripts. It has the logic there.

Reply:
It also turns out that we don’t have git info on the helix machines